### PR TITLE
feature(unlock-app): adding support for universal credit card for locksmith

### PR DESCRIPTION
**Pull request by julien51**
# Description

Locksmith part for the universal credit card support.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/controllers/v2/purchaseController.ts [hunk 1]
```diff
@@ -161,3 +166,122 @@ export const removePaymentMethods: RequestHandler = async (
...
+      destination_network: 'ethereum', // use polygon when applicable?
+      supported_destination_currencies: ['usdc'],
+      supported_destination_networks: ['ethereum'],
+      wallet_addresses: {
+        ethereum: userAddress,
+      },
+    },
+  })
+
+  // TODO: save everything to DB?
```
Comment (julien51) [inline locksmith/src/controllers/v2/purchaseController.ts:262]: I still have to add that part.

File: locksmith/src/controllers/v2/purchaseController.ts [hunk 2]
```diff
@@ -161,3 +166,122 @@ export const removePaymentMethods: RequestHandler = async (
...
+}
+
+/**
+ * Execute the purchase transaction!
+ * @param request
+ * @param response
+ * @returns
+ */
+export const captureOnRamp: RequestHandler = async (request, response) => {
+  const transactionHash = request.query.transactionHash
```
Comment (julien51) [inline locksmith/src/controllers/v2/purchaseController.ts:282]: And that part.

**Review by searchableguy — DISMISSED** [review]

File: locksmith/src/controllers/v2/priceController.ts [hunk 3]
```diff
@@ -41,6 +47,59 @@ export const total: RequestHandler = async (request, response) => {
...
+        return {
+          userAddress: recipient.address,
+          amount: recipient.price.amount,
+          symbol: '$',
+        }
+      }),
+    ],
+  })
+}
+
```
Comment (searchableguy) [inline locksmith/src/controllers/v2/priceController.ts:102]: Rebase after merging the other PR?

Comment (julien51) [reply locksmith/src/controllers/v2/priceController.ts]: yyp!

**Comment by julien51** [comment]
I need to add the openapi description but please PTAL @searchableguy if you can!

File: locksmith/src/controllers/v2/purchaseController.ts [hunk 4]
```diff
@@ -161,3 +169,148 @@ export const removePaymentMethods: RequestHandler = async (
...
+    6
+  )
+
+  const session = await new OnrampSessionResource(stripe).create({
+    transaction_details: {
+      lock_wallet_address: true, // Making sure the user does not change the wallet!
+      source_currency: 'usd', // We only support USDC to USDC
+      destination_currency: 'usdc', // We only support USD to USDC
+      destination_exchange_amount: amount,
+      destination_network: 'ethereum', // use polygon when applicable?
```
Comment (julien51) [inline locksmith/src/controllers/v2/purchaseController.ts:256]: Actually I will add this to the network package for clarity.

**Review by searchableguy — APPROVED** [review]

**Pull request merged**